config/db.js

```diff
@@ -3,22 +3,12 @@ const { db: { uri: databaseURI } } = require('./index');
 const logger = require('../helpers/logger');
 const { gracefulShutdown } = require('../helpers/util');
 
-mongoose.connect(databaseURI, {
-  useNewUrlParser: true,
-  useUnifiedTopology: true,
-  useFindAndModify: false,
-  useCreateIndex: true,
-});
-
-const conn = mongoose.connection;
+const conn = async () => mongoose.connect(databaseURI);
 
-conn.on('open', () => {
-  logger.info('Database Connected');
-});
+conn()
+  .catch((error) => logger.error(error));
 
-conn.on('error', (err) => {
-  logger.error(`Connection error: ${err.message}`);
-});
+module.exports = conn();
```

Comment:
You're calling the function in your export here @Andrew-Paystack, rather than passing the function to be used elsewhere